src/encdatavault_common_plug.c

```diff
@@ -161,14 +164,15 @@ void *get_salt_common(char *ciphertext, int is_pbkdf2)
 	p = strtokm(NULL, "$");
 	cs.algo_id = atoi(p);
 	p = strtokm(NULL, "$");
-	for (i = 0; i < ENC_NONCE_SIZE; i++)
+	for (i = 0; i < ENC_IV_SIZE; i++)
 		cs.iv[i] = (atoi16[ARCH_INDEX(p[2 * i])] << 4) | atoi16[ARCH_INDEX(p[2 * i + 1])];
```

Comment:
Looks like only half of the IV was decoded before, yet somehow this did not matter. Why did it not matter?

Reply:
the second half is not used in the hashing algorithm.